pkg/prometheus/operator_test.go

```diff
@@ -138,3 +138,67 @@ func TestGetNodeAddresses(t *testing.T) {
 		})
 	}
 }
+
+func TestStatefulSetKeyToPrometheusKey(t *testing.T) {
+	cases := []struct {
+		input         string
+		expectedKey   string
+		expectedMatch bool
+	}{
+		{
+			input:         "namespace/prometheus-test",
+			expectedKey:   "namespace/test",
+			expectedMatch: true,
+		},
+		{
+			input:         "namespace/prometheus-test-shard-1",
+			expectedKey:   "namespace/test",
+			expectedMatch: true,
+		},
+		{
+			input:         "allns-z-thanosrulercreatedeletecluster-qcwdmj-0/thanos-ruler-test",
+			expectedKey:   "",
+			expectedMatch: false,
+		},
+	}
+
+	for _, c := range cases {
+		match, key := statefulSetKeyToPrometheusKey(c.input)
+		if c.expectedKey != key {
+			t.Fatalf("Expected prometheus key %q got %q", c.expectedKey, key)
+		}
+		if c.expectedMatch != match {
+			notExp := ""
+			if !c.expectedMatch {
+				notExp = "not "
+			}
+			t.Fatalf("Expected input %sto be matching a prometheus key, but did not", notExp)
+		}
+	}
+}
+
+func TestPrometheusKeyToStatefulSetKey(t *testing.T) {
+	cases := []struct {
+		name     string
+		shard    int
+		expected string
+	}{
+		{
+			name:     "namespace/test",
+			shard:    0,
+			expected: "namespace/prometheus-test",
+		},
+		{
+			name:     "namespace/test",
+			shard:    1,
```

Comment:
Would a shard with -1 value be also a good thing to add as a test case?

Reply:
I don't think so. The code tested here is guarded by code that tests that shards are always >=1.